packages/csvviewer-extension/src/searchprovider.ts

```diff
@@ -4,13 +4,17 @@ import { ISearchProvider, ISearchMatch } from '@jupyterlab/documentsearch';
 import { CSVViewer } from '@jupyterlab/csvviewer';
 import { IDocumentWidget, DocumentWidget } from '@jupyterlab/docregistry';
 import { Signal, ISignal } from '@phosphor/signaling';
+import { Widget } from '@phosphor/widgets';
 
-export class CSVSearchProvider implements ISearchProvider {
+// The type for which canSearchFor returns true
+export type CSVDocumentWidget = IDocumentWidget<CSVViewer>;
```

Comment:
I think you could just use `IDocumentWidget<CSVViewer>` directly.

Reply:
Personally I favor keeping `CSVDocumentWidget`, since it has to be used in 4+ places in the code and is easy to screw up (eg using `DocumentWidget<CSVViewer>` instead).

Can we agree to disagree on this one?

Reply:
Also, it avoids a doubly nested type parameter:

```typescript
export class CSVSearchProvider implements ISearchProvider<IDocumentWidget<CSVViewer>>
```

Reply:
If you are calling it DRY, I guess that makes sense. My argument was based on readability (fewer indirections to figure out what the typename actually refers to), but I see your point.